core-reactive/src/test/groovy/io/micronaut/core/async/propagation/ReactivePropagationSpec.groovy

```diff
@@ -0,0 +1,252 @@
+package io.micronaut.core.async.propagation
+
+import io.micronaut.core.propagation.PropagatedContext
+import io.micronaut.core.propagation.PropagatedContextElement
+import org.reactivestreams.Publisher
+import org.reactivestreams.Subscriber
+import org.reactivestreams.Subscription
+import reactor.core.CorePublisher
+import reactor.core.CoreSubscriber
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import spock.lang.Specification
+
+import static io.micronaut.core.async.propagation.ReactorPropagation.addPropagatedContext
+
+class ReactivePropagationSpec extends Specification {
+
+    def "test context propagation on Publisher methods"() {
```

Comment:
Can you pls create a real-world test where is that needed?

Reply:
Ah I see there is an example in tracing

Reply:
Please take a look at these tests in tracing: https://github.com/micronaut-projects/micronaut-tracing/pull/498